### PR TITLE
Python 3.11 compatibility + Test on Python3.11

### PR DESCRIPTION
**Pull request by dweindl**
* GHA: Run tests on Python3.11
* Fixes related to Python3.11 changes to enums (https://docs.python.org/3/whatsnew/3.11.html#enum)
* Fix unreliable string `is` comparison

### CODE REVIEW

**Comment by dweindl** [comment]
Still some python3.11 wheels missing on pypi (libsbml, h5py, ...)

File: .github/workflows/test_python_ver_matrix.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.8, 3.9, '3.10']
+        python-version: [3.8, 3.9, '3.10', '3.11']
```
Comment (FFroehlich) [inline .github/workflows/test_python_ver_matrix.yml:27]:
```suggestion
        python-version: [3.9, '3.10', '3.11']
```

Comment (dweindl) [reply .github/workflows/test_python_ver_matrix.yml:27]: We wanted to follow https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule, then we should wait another 6 months.

Comment (FFroehlich) [reply .github/workflows/test_python_ver_matrix.yml:27]: ah 👍

File: .github/workflows/test_python_ver_matrix.yml [hunk 2]
```diff
@@ -50,10 +50,18 @@ jobs:
           libhdf5-serial-dev \
           libboost-math-dev
 
+    - name: Extra dependencies
+      # h5py from source
+      run: |
+        if [[ ${{ matrix.python-version }} == "3.11" ]] ; then \
+          pip3 install cython
+        fi
+
```
Comment (dweindl) [inline .github/workflows/test_python_ver_matrix.yml:59]:
```suggestion
```

File: .github/workflows/test_python_ver_matrix.yml [hunk 3]
```diff
@@ -50,10 +50,18 @@ jobs:
...
     # install AMICI
     - name: Build BNGL
       run: |
         scripts/buildBNGL.sh
+
```
Comment (dweindl) [inline .github/workflows/test_python_ver_matrix.yml:64]:
```suggestion
```

**Review by FFroehlich — APPROVED** [review]

**Comment by dweindl** [comment]
Finally works with Python3.11
https://github.com/AMICI-dev/AMICI/actions/runs/3996039619/jobs/6855621483

**Pull request merged**